### ISSUE TITLE
Better handling of HTML tags in translation source strings

### ISSUE THREAD

**Issue by loicteixeira** [wagtail/wagtail#7112]
### Is your proposal related to a problem?

HTML tags are present in source translation strings and can be missing from translated strings

### Describe the solution you'd like

Forbidding the use of HTML tags in source strings might not be feasible, so either:

- Some linting when pulling translated string to make sure no tags are missing
- Setting some magic configuration setting on Transifex, so we get a warning that the tags are missing. If no such setting exist, we could raise the issue with Transifex.

### Additional context

Screenshot | Notes
--- | ---
![HTML Tags_Encapsulate](https://user-images.githubusercontent.com/6834938/116189367-24554d80-a729-11eb-871e-dd40b5adf751.png) | This isn’t as big of an issue as it used to be as HTML Tags are now clickable and embed placeholders (see tooltip)
![HTLM Tags_Missing](https://user-images.githubusercontent.com/6834938/116189442-3cc56800-a729-11eb-82f9-c21b5d7530d2.png)  | However, there is still no warning for missing tags

**Comment by thibaudcolas** [comment]
Hi @loicteixeira, would there be a way for us to write translation strings so they don’t contain the tags?

**Comment by loicteixeira** [comment]
If the only part to translate is the inner html, then yes, the tags could be left off,  
for example change `_('<a href="%(url)">some action name to be translated</a>')`  
to `action_name = _('some action name to be translated'); f"<a href="${url}">${action_name}</a>"`  
or something like that.

However if the tags are within a sentence, I'm afraid we can't because different languages might need to move this link around in the sentence.

That being said, reading the Transifex doc again, they mention a _HTML tags are preserved in the translation_ setting to enable in the [translation checks](https://docs.transifex.com/translation-checks/setting-translation-checks) so I'll investigate this immediately!

**Comment by tomdyson** [comment]
I've changed the setting from this:

![screenshot 2022-05-04 at 09 33 01](https://user-images.githubusercontent.com/15543/166647475-1bc6a5ef-c2cb-492b-af3c-936be5e6a5ed.png)

to this:

![screenshot 2022-05-04 at 09 33 21](https://user-images.githubusercontent.com/15543/166647485-8ef4645f-216f-4e6b-994e-8090a835fd59.png)

**Comment by loicteixeira** [comment]
Perfect thanks. I'll do a few tests and update/close the issue accordingly.

**Comment by lb-** [comment]
@loicteixeira how did you go with checks from changes done in May?

**Comment by loicteixeira** [comment]
Ah! I forgot to comment here after testing 😞 

Unfortunately, enabling the setting did not seem to do anything. So I either did something wrong, or misunderstood what it is supposed to do. I'll give it another go.

FWIW, the rule was off this morning, so I turned it back into a warning, but will investigate further later.

**Comment by lb-** [comment]
thanks for investigating Loic.

**Comment by loicteixeira** [comment]
Tested again today, missing tags now show a warning:
![missing_tag](https://user-images.githubusercontent.com/6834938/216083964-941ccb03-e3d0-4404-aa97-8b775fb3c44f.png)

The placeholder number is wrong, but all 3 tags are `b` tags, so I reckon they can't really tell which one is missing.

I also tested removing a tag, and writing another tag by hand (for example it was `<b>...</b>` and I wrote `<span>...</span>`). Upon save, the span tags are converted to the purple clickable number badge, but they get a different number, and the warning for missing tag is shown:
![renamed_tag](https://user-images.githubusercontent.com/6834938/216084801-df405a79-f4e1-461a-af62-065d13ee1ebe.png)

All those warnings can be dismissed (or is it because I have some admin rights?), but I reckon it's good enough as it is. Closing this then.
